opm/simulators/wells/WellInterfaceEval.cpp

```diff
@@ -342,11 +342,11 @@ assembleControlEqProd_(const WellState& well_state,
         break;
     }
     case Well::ProducerCMode::BHP: {
-        control_eq = bhp - controls.bhp_limit;
+        control_eq = (bhp - controls.bhp_limit)*this->bhpControlScaling();
```

Comment:
From here,  if the bhp_scale_ and bhp_control_scale are different values, how will it affects the tolerance checking?